inc/azure_c_util/interlocked.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef INTERLOCKED
+#define INTERLOCKED
+
+#ifdef __cplusplus
+#include <cstdint>
+#include <cstdbool>
+
+#else
+#include <stdint.h>
+#include <stdbool.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+#include "umock_c/umock_c_prod.h"
+
+MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value);
+MOCKABLE_FUNCTION(, int32_t, interlocked_and, volatile int32_t*, destination, int32_t, value);
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile int16_t*, destination, int16_t, value);
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile int64_t*, destination, int64_t, value);
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
```

Comment:
>int64_t [](start = 20, length = 7)

int8_t

Reply:
(here and in the .md files of course)

---
In reply to: [424024071](https://github.com/Azure/azure-c-util/pull/39#discussion_r424024071) [](ancestors = 424024071)